packages/core/test/helpers/rule.spec.ts

```diff
@@ -64,6 +69,10 @@ describe(`helpers/rule`, () => {
                 { selector: '& &.x:hover' },
                 { selector: '&.y.x' },
                 { selector: '&&' }, // TODO: check if possible
+                { selector: ':not(&) &' },
+                { selector: ':nth-child(5n - 1 of &)' },
+                { selector: ':nth-child(5n - 2 of &, &)' },
+                { selector: ':nth-child(5n - 3 of &, .x, &)' }, // ToDo: check if to remove unrelated nested selectors
```

Comment:
?

Reply:
the result of the the input `:nth-child(5n - 3 of i, .x, i)` is transformed into `:nth-child(5n - 3 of &, .x, &) {}` - the nested `.x` selector should probably be removed. 

However, there are cases where you would want to keep the nested selector: `.mix:is(:mix-state)`. I haven't had time go through it properly and fix. so comment for now.